src/lora.c

```diff
@@ -21,9 +22,11 @@ static void lora_callback(twr_cmwx1zzabz_t* self, twr_cmwx1zzabz_event_t event,
         case TWR_CMWX1ZZABZ_EVENT_SEND_MESSAGE_START:
             twr_log_debug("lora: Sending message");
             voltage_measure(20, false);
+            
             break;
 
         case TWR_CMWX1ZZABZ_EVENT_SEND_MESSAGE_DONE:
+            last_uplink_confirmed = false;  // If the message is sent, consider it as not confirmed unless explicitly confirmed
```

Comment:
I am not sure I follow the logic for setting the last_uplink_confirmed value in this switch-case statement. This variable should be set to true if the uplink was sent as confirmed in lora_send(). In other word, this variable should track the value of the lora_send() confirmed parameter.

Reply:
The lora_send() confirmed parameter will always be true in this implementation.
The only call to lora_send() calls it with "true" as the arg. 

If we're considerign the uplink confirmed flag to be "message sent" (without considering an ack, yet) then, maybe the right functionality would be to set the flag as false first at the start of the application_task switch tree (possibly in the "idle" or the "measuring" state cases)   then set it to true when lora_send() is called?

Reply:
The idea here is to record if the message was sent as confirmed or unconfirmed by the application. You are right that in the current implementation this is hard-coded to confirmed. However, I was planning to update the firmware more to make it possible to select confirmed/unconfirmed uplink via an AT command or by long-pressing the button.

The confirmed flag is meant to be an indication whether the message was sent as confirmed or unconfirmed. The ack flag then records whether a confirmation to the message was received from the network. This ack flag will be always false when the message is sent as unconfirmed. It can be either true or false when the message is sent as confirmed.